lms/services/d2l_api/client.py

```diff
@@ -106,16 +109,35 @@ def group_set_groups(self, org_unit, group_category_id, user_id=None):
 
         return groups
 
-    def course_table_of_contents(self, org_unit) -> List[dict]:
-        """
-        Get a list of modules in the given course.
-
-        https://docs.valence.desire2learn.com/res/content.html#get--d2l-api-le-(version)-(orgUnitId)-content-toc
-        """
-        response = self._api.request(
-            "GET", self._api.api_url(f"/{org_unit}/content/toc", product="le")
-        )
-        return D2LTableOfContentsSchema(response).parse().get("modules", [])
+    def list_files(self, org_unit) -> List[dict]:
+        """Get a list of files and folders for the given `org_unit`."""
+
+        def _find_files(modules):
+            """Recursively find files in modules."""
+            for module in modules:
+                module_files = []
+                for topic in module.get("topics", []):
+                    if topic.get("type") == "File":
+                        module_files.append(
+                            {
+                                "type": "File",
```

Comment:
Spelling every field here as we don't want to expose everything (ie topics/modules).

We could use another marshmallow schema for the output side of this but it seems overkill.